### ISSUE TITLE
Apply a filter across the ticket backlog table, supporting pagination

### ISSUE THREAD

**Issue by ChrisKellaleaMaynard** [aehrc/snomio#270]
**As a:** Terminology Author,  
**I want to:**  specify a multi-field filter on the list of the AMT backlog tickets
**So I can:**  see a refined list of tickets which adhere to the filter's specification, across all pages of the tickets.

---
The built-in filtering of the ticket backlog table only applies its restrictions to the current page of tickets being displayed.  This means tickets which are available in previous or future pages are not included in the final list of tickets displayed.

### Related Issues
* #166 

### Acceptance Criteria
1. [ thing on-screen and the behaviour it exhibits ]

### Testing
1. [ test case ]

**Comment by kylynnloi** [comment]
@clinnygee - looking good!

- Priority, Schedule, Release, Status, Assignee - can we make it so you can filter for multiple values (always ANY)
- Task - i would like to be able to search on a task number?

**Comment by clinnygee** [comment]
@kylynnloi by the task one do you mean have it as a search bar instead of a dropdown?

**Comment by kylynnloi** [comment]
@clinnygee yep - i think search bar would be better instead.... i was only wondering if its both, but i'm not sure of the value...

**Comment by ChrisKellaleaMaynard** [comment]
A couple points found during review:

1. Since the most recent import of the tickets from Jira, the Iterations have all been cleared.  Please add a few Iterations for testing.
2. There are no Priority values.  These should always be available in the solution, regardless of whether the tickets have recently been transferred from Jira  This may require a change to the Jira import process to ensure the values aren't cleared.
3. Can you please add the ability to sort by **Schedule**.
4. The filtering by **Date Created** does not work properly.  If you set the date to the actual creation date of the ticket, the filter works (I was targeting a ticket assigned to me, created 19/3/2014).  If you set the date to a range including the ticket's creation date (e.g. 18/3/14 - 20/3/14), it works.  However if you select "Before Date" and "01/01/2015", no tickets appear even though there are several in March 2014 at a minimum.  The same occurs for "Date After" and "01/01/2014", and "Date is not".  The same behaviour occurs for new tickets where the **Date Created** is set internally by Snomio.
5. There is no way to filter to tickets that aren't assigned.  I know this was raised before (possibly by Kylynn) but I can see it's still not resolved yet this is a key requirement for the users: they will select the iteration and then filter down to the ones which have yet to be assigned so they can select one to work on.
6. After applying a sort to the backlog, if you click _Clear_ then the list is reset (no sorting/filtering).  If you click _Clear_ again, the list is displayed again, no changes.  If you click _Clear_ again, the list is emptied.  The selection of any sorting/filtering causes the tickets to be displayed again.
7. The "multiple rule" option for selecting labels doesn't seem to work: I select "Match Any" from the top of the options, then "FredIT" from the Label Picker, and _Apply_ - and the correct list is displayed.  I then return to the filtering and click _Add Rule_, select "SAS", and the "FredIT" is replaced by "SAS" as the first rule, and the second rule becomes "Any".
8. The ticket list is currently 2508 pages long.  If I set the **Schedule** filter to all of the available **Schedule** values, there are 2363 pages.  Why are 145 pages removed, if the **Schedule** can only be one of these values in the filter list?
